server/src/main/java/no/fractal/socket/send/AbstractMessage.java

```diff
@@ -1,69 +1,82 @@
 package no.fractal.socket.send;
 
-import no.fractal.socket.StreamUtil;
 import no.fractal.util.StreamUtils;
-import org.json.JSONArray;
 import org.json.JSONObject;
 
-
 import java.io.*;
 import java.nio.file.Files;
-import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.Map;
 
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
 public abstract class AbstractMessage {
 
     private final String messageType;
-    private final ArrayList<Segment> segments;
-
+    private final Map<String, Segment> segments;
+    private JsonObject body = new JsonObject();
 
-    protected AbstractMessage(String messageType
-    ) {
+    protected AbstractMessage(String messageType) {
         this.messageType = messageType;
-        this.segments = new ArrayList<>();
+        this.segments = new LinkedHashMap<String, Segment>();
```

Comment:
e det ikke enklere og ha segment navnet som et felt i seg objektet. da kan man bare ha en liste og logikken blir enklere. Det funke her siden man ikke ska oppdater noen av verdian da har en map vert ber

Reply:
Da må man jo alltid itterere listen for å hente ut  segments da? og hvordan er logikken enklere da?

Reply:
tenkte siden at man ikke noensteds henter ut verdiene så dataen itereres ut uansett. å vil derfor være mere effektive å lagre i en liste men, ikke så falig